### PR TITLE
Leverage type inference for generics

### PR DESCRIPTION
**Pull request by elsazac**
Use <> for type inference wherever applicable

### CODE REVIEW

File: bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/menu/util/MenuAdapterFactory.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ public boolean isFactoryForType(Object object) {
 	 * <!-- end-user-doc -->
 	 * @generated
 	 */
-	protected MenuSwitch<Adapter> modelSwitch = new MenuSwitch<Adapter>() {
+	protected MenuSwitch<Adapter> modelSwitch = new MenuSwitch<>() {
```
Comment (akurtakov) [inline bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/menu/util/MenuAdapterFactory.java]: This is generated code so shouldn't be manually edited. @merks should there be some update in EMF itself or regeneration should help?

Comment (merks) [reply bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/menu/util/MenuAdapterFactory.java]: The generator uses the configured cleanup actions which would make this change, but I it doesn't apply such an action if the generated file is unchanged by generation.  So better would be to do a manual cleanup of the entire project and then regenerate the model(s).

**Comment by merks** [comment]
Please don't change generated files, i.e., files with ` * @generated` in the Javadoc.

I will apply the cleanup to projects with models in them and regenerate them to ensure they remain consistent.

**Comment by merks** [comment]
FYI, I've done a general cleanup and regeneration via this PR:

https://github.com/eclipse-platform/eclipse.platform.ui/pull/1465

**Review by merks — CHANGES_REQUESTED** [review]
Please remove the changes to generated files.  Those are handled by this PR:

https://github.com/eclipse-platform/eclipse.platform.ui/pull/1465

**Comment by elsazac** [comment]
Thanks for the review @akurtakov @merks .
I reverted all the changes in `org.eclipse.e4.ui.model.workbench` bundle, assuming that's where all the generated code go ( I never knew they were generated code). Let me know if this is fine.

**Review by merks — APPROVED** [review]
It looks fine now.  Thanks.  😄

**Pull request merged**